pandas/io/parquet.py

```diff
@@ -260,8 +260,13 @@ def read_parquet(path, engine='auto', columns=None, **kwargs):
 
     Parameters
     ----------
-    path : string
-        File path
+    path : various
+        Either a path to a file (a :class:`python:str`,
+        :class:`python:pathlib.Path`,
+        or :class:`py:py._path.local.LocalPath`),
+        URL (including http, ftp, and S3 locations), or any object
+        with a ``read()`` method (such as an open file or
```

Comment:
``read()`` and ``seek()`` are required; I'd rather describe it as a file-like object instead of mentioning the explicit methods that need to be implemented.

Reply:
Agree, file-like is more clear. Just for the record I took that description from https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#basic, and replaced StringIO with ByteIO